src/autogen.ml

```diff
@@ -9,42 +9,84 @@ let mk_mapper file =
 let mk_output dir file =
   Filename.concat dir (file ^ ".ml")
 
+let mk_meta_json dir =
+  Filename.concat dir "meta.json"
+
 let exec ?(fatal=false) cmd error_msg =
-  if Sys.command cmd <> 0 then
+  let exit_value = Sys.command cmd in
+  if exit_value <> 0 then
     begin
       Printf.eprintf "%s" error_msg;
       if fatal then Printf.printf "Stopping the program.";
-      exit 1
+      exit exit_value
     end
 
 let remove_trailing_whitespaces output =
   let error_msg =
     Printf.sprintf "Could not remove trailing whitespaces in file %s." output in
-  exec ("sed -Ei 's/[ \\t]+$//' " ^ output) error_msg
+  exec ("sed -i.bak -E 's/[ \t]+$//' " ^ output) error_msg
 
 let insert_line_between_each_functions output template_fill =
+  let cmd = Printf.sprintf "sed -i.bak -E '/%s/G' %s" template_fill output in
+  exec cmd "Could not insert line breaks."
+
+let change_template_fill output template_fill =
   let default_fill = "Replace this string by your implementation." in
   let cmd =
-    Printf.sprintf "sed -i 's/%s/%s/; /%s/G' %s"
-    default_fill template_fill template_fill output in
+    Printf.sprintf "sed -i.bak -E 's/%s/%s/' %s"
```

Comment:
`sed` is used many times in this module: please introduce a function to encapsulate low-level details properly.